pos-print/src/main/java/com/clouway/pos/print/PosPrintService.java

```diff
@@ -39,6 +40,10 @@ public static void main(String[] args) {
       new PersistentModule(client, commandCLI.dbName())
     );
 
+    BackgroundReceiptPrinter backgroundPrinter = injector.getInstance(BackgroundReceiptPrinter.class);
```

Comment:
You can use guava services about this as it will enforce lifecycle (start/stop) of the printing service.

Reply:
Implemented Guava services for the background thread.